db/1.9/modules/ROOT/pages/database-connector-select.adoc

```diff
@@ -0,0 +1,184 @@
+
+The advantages of using input parameters to configure the `WHERE` clause in a SQL `SELECT` statement are that it makes the query immune to SQL injection attacks and enables optimizations that are not possible otherwise improving the application performance.
+
+For security reasons, do not directly write `<db:sql>SELECT * FROM PLANET WHERE name = #[payload] </db:sql>`.
+
+DataSense is available for the operation's input and output. Database Connector analyzes the query and automatically calculates the structure of the query's output by analyzing the projection section of the SQL statement. At the same time, by comparing the conditions in the `WHERE` clause to the table structure, it generates DataSense input to help you build the DataWeave script that generates the input parameters.
+
+In the following example, you supply input parameters as key-value pairs embedding a DataWeave script. The keys use the colon character (`:`) to reference a parameter value by name:
+
+. In your Studio flow, select the *Select* operation.
+. In the operation configuration screen, set the *SQL Query Text* field to `SELECT * FROM PLANET WHERE name = :name`.
+. Set the *Input Parameters* field to `{'name' : payload}`.
+
+The following screenshot shows the *Input Parameters*  configuration:
+
+.Input Parameters configuration
+image::database-select-operation-2.png[Input Parameters configuration in Studio]
+
+In the XML editor, the `<db:input-parameters>` configuration looks like this:
+[source,xml,linenums]
+----
+<flow name="DatabaseFlow">
+  <db:select config-ref="dbConfig">
+    <db:sql>SELECT * FROM PLANET WHERE name = :name</db:sql>
+    <db:input-parameters>
+      #[{'name' : payload}]
+    </db:input-parameters>
+  </db:select>
+</flow>
+----
+
+[NOTE]
+====
+*Since version 1.4.0 (escaping colons)*:
+If you need to use the colon character (`:`) in your SQL query, you can escape it
+by putting a backslash before it. This is useful when using PostgreSQL type
+casting, which requires two colons before the type you are casting to, for example:
+
+`<db:sql>SELECT price\:\:float8 FROM PRODUCT</db:sql>`
+====
+
+[NOTE]
+====
+*Since version 1.8.0 (SQL Casting for PostgreSQL and Snowflake)*:
+Database Connector now accepts SQL Casting PostgreSQL and Snowflake syntax's, through the double colon (::) expression without the need to escape each colon (this feature doesn't affect the escaping colons behaviour). For example:
+
+`<db:sql>SELECT MAX(modified_date)::DATE FROM sales</db:sql>`
+====
+
+
+== Configure Dynamic Queries
+
+When you need to parameterize not only the `WHERE` clause, but also parts of the query itself, for example queries that need to compare tables that depend on a condition, or complex queries for which the project table columns need to vary, you can configure dynamic queries.
```

Comment:
This is a LOT of modification for one sentence. Consider making it more than one. Or use parens for some of the modification, to prevent confusion around all the commas:
```suggestion
When you need to parameterize not only the `WHERE` clause but also parts of the query itself (for example, queries that compare tables that depend on a condition, or complex queries for which the project table columns vary), you can configure dynamic queries.
```